src/ucp/tag/rndv.c

```diff
@@ -818,6 +839,8 @@ ucp_rndv_recv_start_get_pipeline(ucp_worker_h worker, ucp_request_t *rndv_req,
     offset = 0;
     while (offset != size) {
         length = ucs_min(size - offset, max_frag_size);
+        length = ucp_rndv_adjust_zcopy_length(min_zcopy, max_frag_size, 0,
```

Comment:
maybe unite line 841 into the function ucp_rndv_adjust_zcopy_length ?

Reply:
yes, it would be great
but using this function for `ucp_rndv_progress_rma_get_zcopy` in [src/ucp/tag/rndv.c:543](https://github.com/openucx/ucx/blob/dc6d0b738738e77e6d42275a694af0df582aae28/src/ucp/tag/rndv.c#L543) doesn't allow this, since it calculates `length` another way

Reply:
> but using this function for `ucp_rndv_progress_rma_get_zcopy` in [src/ucp/tag/rndv.c:543](https://github.com/openucx/ucx/blob/dc6d0b738738e77e6d42275a694af0df582aae28/src/ucp/tag/rndv.c#L543) doesn't allow this, since it calculates `length` another way

i think ucs_min with max_get_zcopy is applicable anyway

Reply:
> i think ucs_min with max_get_zcopy is applicable anyway

yes, we could do it
done, will push after closing [another discussion](https://github.com/openucx/ucx/pull/5273/files#r439962008)